### ISSUE TITLE
Define dependencies in composer.json

### ISSUE THREAD

**Issue by grasmash** [acquia/lightning#94]
We're currently using composer to manage dependencies as per the model set forth in github.com/drupal-composer/drupal-project.

However, using this does not allow Lightning to pull in its dependencies. 

Here's a headstart:
````
{
    "name": "acquia/lightning",
    "description": "The intent of Lightning is to provide a lightweight framework for building working solutions in Drupal.",
    "type": "drupal-profile",
    "repositories": [
        {
            "type": "composer",
            "url": "https://packagist.drupal-composer.org"
        }
    ],
    "require": {
        "composer/installers": "^1.0.20",
        "cweagans/composer-patches": "~1.0",
        "drupal/embed": "8.0-1.0#26265b5437bb0f1dcb916a6a1169d53382a23646",
        "drupal/entity_embed": "8.0-1.0#9760c0a790bccb30d7e15815fdfa6d1794453d3a",
        "drupal/media_entity": "8.0-1.0#f8decb81965037b05c51ce8c16eae8c37256e7a0",
        "drupal/media_entity_embeddable_video": "8.0-1.0#f7e51c2cccd875e1fd09d018c11fd50c93a71507",
        "drupal/media_entity_instagram": "8.0-1.0#75e41cbae35f260fa0e3e76f90562e26b4dc162d",
        "drupal/media_entity_twitter": "8.0-1.0#179e19e159f7df2172316030fc7c110b6f0d3efd",
        "drupal/media_entity_image": "8.0-1.0#f17384aeba3c8c00bbc57bfa8fa68529992873ce",
        "drupal/ctools": "8.0-3.0#e27c2663027a4f90dffb39f7fd6ed8a92e5d61d4",
        "drupal/layout_plugin": "8.0-1.0#993c829961b5b304a0094d7cd33f014edccf859e",
        "drupal/panels": "8.0-3.0#a458d139e2e9e35c37048cb30dc682087593bd4d",
        "drupal/page_manager": "8.0-1.0#3f15c87fec006e78954bb3e984cb80afeea4f5b2",
        "drupal/acquia_connector": "8.0-1.0#302fc469eb860892fc90403b5d758eb50a715f77",
        "drupal/metatag": "1.0-beta3",
        "drupal/core": "8.*"
    },
    "minimum-stability": "dev",
    "prefer-stable": true,
    "extra": {
        "patches": {
            "drupal/ctools": {
                "Customize views block settings on each block instance": "https://www.drupal.org/files/issues/2607594-20.patch"
            }
        }
    }
}
````

**Comment by grasmash** [comment]
Clearly, I haven't included any of the JS libraries in this. They'll need to be managed by a front end dependency manager like Grunt or Bower.

**Comment by balsama** [comment]
Thanks Matt. This is something we're definitely interested in integrating.

There was some question about overhead in maintaining this + the make-based workflow that D.O requires when we initially looked into this. (We're already rewriting our `*.make.yml` into `*.make` files for D.O)... But we're very interested in supporting a composer-based build too. Stay tuned.

**Comment by grasmash** [comment]
@balsama Ah ok. This is currently a blocker for the PS team. I'm working on two things at the moment, not sure if either will pan out:

* A drush command that converts make files to composer.json.
* A plugin for drupal-composer that will read make files.

At present, you could speed up your own maintenance by using the `drush make-convert` command that ships with drush 8.x to convert from make to yml automatically.

**Comment by balsama** [comment]
> At present, you could speed up your own maintenance by using the drush make-convert command that ships with drush 8.x to convert from make to yml automatically.

We want to go the other way :) (yml -> make) That's why we have the prepmake script.

**Comment by balsama** [comment]
See: https://www.drupal.org/node/2693829

**Comment by grasmash** [comment]
Awesome! Spoke briefly with Adam H about the strategy here. Test
integration and get back to you . Nice work!
On Fri, Apr 1, 2016 at 12:36 AM Adam Balsam <notifications@github.com>
wrote:

> Closed #94 <https://github.com/acquia/lightning/issues/94>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/acquia/lightning/issues/94#event-610783064>
>
-- 
*Matthew Grasmick** | Technical Team Lead* | *Acquia*
<http://www.acquia.com/>
*E: matthew.grasmick@acquia.com <matthew.grasmick@acquia.com>* | *C*: (443)
865-0353 | *S*: grasmash

Acquia Identified as a ​leader in new Gartner WCM Magic Quadrant
<https://www.acquia.com/gartner-magic-quadrant>
Follow @Acquia on Twitter <https://twitter.com/Acquia>
